pkg/nsx/services/subnet/subnet.go

```diff
@@ -459,3 +461,148 @@ func (service *SubnetService) UpdateSubnetSet(ns string, vpcSubnets []*model.Vpc
+		case "DHCP_RELAY":
+			subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeRelay)
+		default:
+			subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated)
+		}
+	} else {
+		subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated)
+	}
+
+	// Map AdvancedConfig
+	subnetCR.Spec.AdvancedConfig.EnableVLANExtension = true
+	if nsxSubnet.AdvancedConfig != nil {
+		if nsxSubnet.AdvancedConfig.ConnectivityState != nil {
+			connectivityState := *nsxSubnet.AdvancedConfig.ConnectivityState
+			switch connectivityState {
+			case "CONNECTED":
+				subnetCR.Spec.AdvancedConfig.ConnectivityState = v1alpha1.ConnectivityStateConnected
+			case "DISCONNECTED":
+				subnetCR.Spec.AdvancedConfig.ConnectivityState = v1alpha1.ConnectivityStateDisconnected
+			}
+		}
+	}
+}
+
+// MapNSXSubnetStatusToSubnetCRStatus maps NSX subnet status to Subnet CR status
+func (service *SubnetService) MapNSXSubnetStatusToSubnetCRStatus(subnetCR *v1alpha1.Subnet, statusList []model.VpcSubnetStatus) {
+	// Clear existing status fields
+	subnetCR.Status.NetworkAddresses = subnetCR.Status.NetworkAddresses[:0]
+	subnetCR.Status.GatewayAddresses = subnetCR.Status.GatewayAddresses[:0]
+	subnetCR.Status.DHCPServerAddresses = subnetCR.Status.DHCPServerAddresses[:0]
+
+	// Set the shared flag to true for shared subnets
+	if _, ok := subnetCR.Annotations[common.AnnotationAssociatedResource]; ok {
+		subnetCR.Status.Shared = true
+	}
+
+	// Map status fields from NSX subnet status
+	for _, status := range statusList {
+		if status.NetworkAddress != nil {
+			subnetCR.Status.NetworkAddresses = append(subnetCR.Status.NetworkAddresses, *status.NetworkAddress)
+		}
+		if status.GatewayAddress != nil {
+			subnetCR.Status.GatewayAddresses = append(subnetCR.Status.GatewayAddresses, *status.GatewayAddress)
+		}
+		// DHCPServerAddress is only for the subnet with DHCP enabled
+		if status.DhcpServerAddress != nil {
+			subnetCR.Status.DHCPServerAddresses = append(subnetCR.Status.DHCPServerAddresses, *status.DhcpServerAddress)
+		}
+	}
+}
+
+// BuildSubnetCR creates a Subnet CR object with the given parameters
+func (service *SubnetService) BuildSubnetCR(ns, subnetName, vpcFullName, associatedName string, nsxSubnet *model.VpcSubnet) *v1alpha1.Subnet {
+	// Create the Subnet CR
+	subnetCR := &v1alpha1.Subnet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      subnetName,
+			Namespace: ns,
+			Annotations: map[string]string{
+				common.AnnotationAssociatedResource: associatedName,
+			},
+		},
+		Spec: v1alpha1.SubnetSpec{
+			VPCName: vpcFullName,
+		},
+	}
+
+	// Initialize subnetCR from nsxSubnet if available
+	service.MapNSXSubnetToSubnetCR(subnetCR, nsxSubnet)
+	return subnetCR
+}
+
+// IsDefaultProject checks if the given project is a default project
+func (service *SubnetService) IsDefaultProject(orgID, projectID string) (bool, error) {
```

Comment:
Maybe rename it to IsDefaultNSXProject to distinguish with the default project of supervisor? Or at least add a comment to distinguish them